### ISSUE TITLE
Rename `IReaderBase.peak()` to `peek()`

### ISSUE THREAD

**Issue by knutwannheden** [jsonjoy-com/util#6]
TLDR; Rename `peak()` to `peek()` in `IReaderBase` and classes that extend it. Still leave `peak()` implementation, but alias it to `peek()` and add `@deprecated` notice for it.

---

I noticed that `IReaderBase` and subtypes define a method called `peak()`, which probably more appropriately would be called `peek()`. Although I don't know what the backwards compatibility strategy is for this library, so I thought I would just create an issue for this rather than a PR. Also, I am aware that there are already downstream libraries like `json-pack` which build on this. Maybe it would make most sense to just deprecate `peak()`.
